src/legacy/core_plugins/embeddable_api/public/np_ready/public/tests/container.test.ts

```diff
@@ -17,46 +17,62 @@
+import { CONTACT_CARD_EMBEDDABLE } from '../lib/test_samples/embeddables/contact_card/contact_card_embeddable_factory';
+import { SlowContactCardEmbeddableFactory } from '../lib/test_samples/embeddables/contact_card/slow_contact_card_embeddable_factory';
+import { HELLO_WORLD_EMBEDDABLE_TYPE } from '../lib/test_samples/embeddables/hello_world/hello_world_embeddable';
+import { HelloWorldEmbeddableFactory } from '../lib/test_samples/embeddables/hello_world/hello_world_embeddable_factory';
+import { HelloWorldContainer } from '../lib/test_samples/embeddables/hello_world_container';
+import {
   ContactCardEmbeddableInput,
   ContactCardEmbeddableOutput,
-  SlowContactCardEmbeddableFactory,
-  HELLO_WORLD_EMBEDDABLE_TYPE,
-  HelloWorldEmbeddableFactory,
-} from '../test_samples/index';
-import { isErrorEmbeddable, EmbeddableOutput, EmbeddableFactory } from '../embeddables';
-import { ContainerInput } from './i_container';
-import { ViewMode } from '../types';
+  ContactCardEmbeddable,
+} from '../lib/test_samples/embeddables/contact_card/contact_card_embeddable';
 import {
-  FilterableEmbeddableInput,
-  FilterableEmbeddable,
-} from '../test_samples/embeddables/filterable_embeddable';
-import { ERROR_EMBEDDABLE_TYPE } from '../embeddables/error_embeddable';
-import { PanelNotFoundError } from './panel_not_found_error';
-
-jest.mock('ui/new_platform');
-
-const embeddableFactories = new Map<string, EmbeddableFactory>();
-embeddableFactories.set(FILTERABLE_EMBEDDABLE, new FilterableEmbeddableFactory());
-embeddableFactories.set(CONTACT_CARD_EMBEDDABLE, new SlowContactCardEmbeddableFactory());
-embeddableFactories.set(HELLO_WORLD_EMBEDDABLE_TYPE, new HelloWorldEmbeddableFactory());
+  FilterableContainer,
+  FilterableContainerInput,
+} from '../lib/test_samples/embeddables/filterable_container';
+import { coreMock } from '../../../../../../../core/public/mocks';
+import { testPlugin } from './test_plugin';
+import { of } from './helpers';
 
 async function creatHelloWorldContainerAndEmbeddable(
   containerInput: ContainerInput = { id: 'hello', panels: {} },
   embeddableInput = {}
 ) {
-  const container = new HelloWorldContainer(containerInput, embeddableFactories);
+  const coreSetup = coreMock.createSetup();
+  const coreStart = coreMock.createStart();
+  const { setup, doStart } = testPlugin(coreSetup, coreStart);
+  const start = doStart();
+
+  const filterableFactory = new FilterableEmbeddableFactory();
+  const slowContactCardFactory = new SlowContactCardEmbeddableFactory({
+    execAction: start.executeTriggerActions,
+  });
+  const helloWorldFactory = new HelloWorldEmbeddableFactory();
+
+  setup.registerEmbeddableFactory(filterableFactory.type, filterableFactory);
+  setup.registerEmbeddableFactory(slowContactCardFactory.type, slowContactCardFactory);
+  setup.registerEmbeddableFactory(helloWorldFactory.type, helloWorldFactory);
+
+  const container = new HelloWorldContainer(containerInput as any, {
```

Comment:
can you get rid of `as any` casting of containerInput?

Reply:
Yes, removed, probably needed it while refactoring.